packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
+    const allowedContexts = [FrameContexts.content];
+    describe('Testing pages.fullTrust.enterFullScreen function', () => {
+      it('pages.fullTrust.enterFullScreen should not allow calls before initialization', () => {
+        expect(() => pages.fullTrust.enterFullscreen()).toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`pages.fullTrust.enterFullScreen should successfully enter fullscreen when initialized with ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            pages.fullTrust.enterFullscreen();
+            const enterFullscreenMessage = utils.findMessageByFunc('enterFullscreen');
+            expect(enterFullscreenMessage).not.toBeNull();
+          });
+        } else {
+          it(`pages.fullTrust.enterFullScreen does not allow calls from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            expect(() => pages.fullTrust.enterFullscreen()).toThrowError(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
+
+    describe('Testing pages.fullTrust.exitFullscreen function', () => {
+      it('pages.fullTrust.exitFullscreen should not allow calls before initialization', () => {
+        expect(() => pages.fullTrust.exitFullscreen()).toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`pages.fullTrust.exitFullscreen should successfully exit fullscreen when initialized with ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            pages.fullTrust.exitFullscreen();
+            const enterFullscreenMessage = utils.findMessageByFunc('exitFullscreen');
+            expect(enterFullscreenMessage).not.toBeNull();
+          });
+        } else {
+          it(`pages.fullTrust.exitFullscreen does not allow calls from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            expect(() => pages.fullTrust.exitFullscreen()).toThrowError(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
+
+    describe('Testing pages.fullTrust.isSupported function', () => {
+      it('pages.fullTrust.isSupported should return false if the runtime says pages.fullTrust is not supported', () => {
+        utils.setRuntimeConfig({ apiVersion: 1, supports: { pages: {} } });
```

Comment:
Can you please add a test that `pages.fullTrust.isSupported` returns false when we set the runtime config to `supports: {}`?

Reply:
Added in the latest commit